### ISSUE TITLE
Bug in the way w34highchartsSearchX.py calculates max() values

### ISSUE THREAD

**Issue by thagorastos** [steepleian/weewx-Weather34#18]
So, I think I detected a bug in the code. In my setup, I've programmed a custom driver for transforming the instrument readings for weewx and w34highchartsSearchX.py seems to crash weewx. I've purposely left the last line on the log there as a reminder of how my readings are transformed (of course this is just a debug printout from my custom driver). It should be an easy fix. Thanks!!!

`Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine: Caught unrecoverable exception in generator 'weewx.cheetahgenerator.CheetahGenerator'
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****  max() arg is an empty sequence
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****  Traceback (most recent call last):
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 196, in run
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      obj.start()
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 281, in start
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      self.run()
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 149, in run
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      ngen = self.generate(gen_dict[section_name], self.gen_ts)
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 219, in generate
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      ngen += self.generate(section[subsection], gen_ts)
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 219, in generate
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      ngen += self.generate(section[subsection], gen_ts)
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 308, in generate
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      default_binding)
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 379, in _getSearchList
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      searchList += obj.get_extension_list(timespan, db_lookup)
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/w34highchartsSearchX.py", line 1848, in get_extension_list
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      sle_dict['wr' + _suffix + ("" if source[0] == 'windSpeed' else "Gust")] = self.calcWindRose(timespan, db_lookup, period, source[0], source[1])
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/w34highchartsSearchX.py", line 1580, in calcWindRose
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      maxSpeed = _max(speed_vec_vt[0])
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/w34highchartsSearchX.py", line 104, in _max
**Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****      return max([v for v in values if v is not None])
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****  ValueError: max() arg is an empty sequence
Dec 18 01:55:41 meteo weewx[384] ERROR weewx.reportengine:         ****  Generator terminated**
Dec 18 01:55:41 meteo weewx[384] INFO weewx.reportengine: Copied 0 files to /var/www/html/weewx/weather34/w34highcharts
**Dec 18 01:55:42 meteo weewxd: Microbit: {'pressure': 1023.0, 'outHumidity': 75.0, 'soilTemp1': 17.55, 'soilMoist1': 8.0, 'rain': 0.83, 'windDir': 0.0, 'outTemp': 18.78, 'windSpeed': 0.0}**
`

**Comment by balloonhotair** [comment]
Since the function in question is from the original code done by Gary over 4 years ago and there has never been an issue before, would mean to me the following. Preventing the error is easy with an if statement and the dev branch already has this change but the real problem is the list should not be empty which indicates a larger issue with the data collection part of the weewx system or the system drivers or code that populates the database.

**Comment by thagorastos** [comment]
I did fix it with an if statement. What worries me is the data collection part you are mentioning; maybe I should check my DB more closely. The custom driver I programmed always provides a value for windspeed and I've added additional checks to make sure that only whole set of validated readings are fed to WeeWx. Anyway, since this fix is already in the dev branch, there is no need for this report.

Thank you for your time!
